game-app/game-core/src/test/java/games/strategy/triplea/ai/pro/ProPurchaseAiTest.java

```diff
@@ -0,0 +1,48 @@
+package games.strategy.triplea.ai.pro;
+
+import static games.strategy.triplea.delegate.MockDelegateBridge.newDelegateBridge;
+import static org.mockito.Mockito.when;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.engine.player.PlayerBridge;
+import games.strategy.triplea.delegate.PurchaseDelegate;
+import games.strategy.triplea.settings.ClientSetting;
+import games.strategy.triplea.xml.TestMapGameData;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.sonatype.goodies.prefs.memory.MemoryPreferences;
+
+public class ProPurchaseAiTest {
+
+  @Test
+  public void testShouldSaveUpForAFleet() {
+    final ProAi proAi = new ProAi("Test Name", "Test Player Label");
+    final GameData gameData = TestMapGameData.VICTORY_TEST_SHOULD_SAVE_UP_FOR_A_FLEET.getGameData();
+    final GamePlayer gamePlayer = gameData.getPlayerList().getPlayerId("Italians");
+    final IDelegateBridge testBridge = newDelegateBridge(gamePlayer);
+    final PurchaseDelegate purchaseDelegate = (PurchaseDelegate) gameData.getDelegate("purchase");
+    final PlayerBridge playerBridgeMock = Mockito.mock(PlayerBridge.class);
+
+    purchaseDelegate.setDelegateBridgeAndPlayer(testBridge);
+    proAi.initialize(playerBridgeMock, gamePlayer);
+    ClientSetting.setPreferences(new MemoryPreferences());
```

Comment:
Hmm this looks like it should be put into a setup function. Something along the lines of this:
```java
class ProPurchaseAiTest {
  private ProAi proAi;
  private GameData gameData;
  private PurchaseDelegate purchaseDelegate;
  private GamePlayer gamePlayer;

  @BeforeEach
  void setUp() {
    proAi = new ProAi("Test Name", "Test Player Label");
    gameData = TestMapGameData.VICTORY_TEST_SHOULD_SAVE_UP_FOR_A_FLEET.getGameData();
    gamePlayer = gameData.getPlayerList().getPlayerId("Italians");
    final IDelegateBridge testBridge = newDelegateBridge(gamePlayer);
    purchaseDelegate = (PurchaseDelegate) gameData.getDelegate("purchase");
    final PlayerBridge playerBridgeMock = Mockito.mock(PlayerBridge.class);

    purchaseDelegate.setDelegateBridgeAndPlayer(testBridge);
    proAi.initialize(playerBridgeMock, gamePlayer);
    ClientSetting.setPreferences(new MemoryPreferences());

    when(playerBridgeMock.getGameData()).thenReturn(gameData);
  }

  @Test 
  void test() {
    Assertions.assertDoesNotThrow(
        () -> proAi.purchase(false, 20, purchaseDelegate, gameData, gamePlayer));
  }
}
```

But I'm honestly not sure if this separation really makes things better. Maybe you can come up with a better split.

Reply:
pulled the setup into a separate private method